xarray/test/test_dataset.py

```diff
@@ -323,6 +323,16 @@ def test_attr_access(self):
         self.assertEqual(ds.attrs['tmin'], -999)
         self.assertDataArrayIdentical(ds.tmin, ds['tmin'])
 
+    def test_dir(self):
+        ds = create_test_data()
+        # add a numbered key to ensure this doesn't break dir
+        ds[5] = list(ds.keys())[0]
```

Comment:
Can you simply create the key directly, e.g., `ds[5] = 'foo'`? It's not clear to me why you're using indexing here.